components/dashboard/src/start/WorkspaceLogs.tsx

```diff
@@ -4,8 +4,81 @@
  * See License-AGPL.txt in the project root for license information.
  */
 
-export default function WorkspaceLogs() {
-  return <pre className="m-6 p-4 h-72 w-11/12 lg:w-3/5 flex-shrink-0 rounded-xl bg-gray-100">
-    ... logs ...
-  </pre>;
+import EventEmitter from 'events';
+import React from 'react';
+import { Terminal, ITerminalOptions, ITheme } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit'
+import 'xterm/css/xterm.css';
+import { DisposableCollection } from '@gitpod/gitpod-protocol';
+
+export interface WorkspaceLogsProps {
+  logsEmitter: EventEmitter;
+}
+
+export interface WorkspaceLogsState {
+}
+
+export default class WorkspaceLogs extends React.Component<WorkspaceLogsProps, WorkspaceLogsState> {
+  protected xTermParentRef: React.RefObject<HTMLDivElement>;
+  protected terminal: Terminal | undefined;
+  protected fitAddon: FitAddon | undefined;
+  
+  constructor(props: WorkspaceLogsProps) {
+    super(props);
+    this.xTermParentRef = React.createRef();
+  }
+
+  private readonly toDispose = new DisposableCollection();
+  componentDidMount() {
+    const element = this.xTermParentRef.current;
+    if (element === null) {
+      return;
+    }
+    const theme: ITheme = {
+      // background: '#F5F5F4',
+    };
+    const options: ITerminalOptions = {
+      cursorBlink: false,
+      disableStdin: true,
+      fontSize: 14,
+      // fontWeight: 'normal',
+      theme,
+      scrollback: 9999999,
+    };
+    this.terminal = new Terminal(options);
+    this.fitAddon = new FitAddon();
+    this.terminal.loadAddon(this.fitAddon);
+    this.terminal.open(element);
+    this.props.logsEmitter.on('logs', logs => {
+      if (this.fitAddon && this.terminal && logs) {
+        this.terminal.write(logs);
+      }
+    });
+    this.toDispose.push(this.terminal);
+    this.fitAddon.fit();
+
+    // Fit terminal on window resize (debounced)
+    let timeout: NodeJS.Timeout | undefined;
+    const onWindowResize = () => {
+      clearTimeout(timeout!);
+      timeout = setTimeout(() => this.fitAddon!.fit(), 20);
+    };
+    window.addEventListener('resize', onWindowResize);
+    this.toDispose.push({
+      dispose: () => {
+        clearTimeout(timeout!);
+        window.removeEventListener('resize', onWindowResize);
+      }
+    });
+  }
+
+  componentWillUnmount() {
+    this.toDispose.dispose();
+  }
+
+  render() {
+    return <div className="mt-6 h-72 w-11/12 lg:w-3/5 rounded-xl bg-black p-6">
```

Comment:
**question**: Could we remove width related classes here so that this defaults to a fixed width? Ideally we could use a smaller width for smaller viewports but responsive layout is out of the scope for this milestone. This is minor though, we can leave this as is.

Reply:
@gtsiolis Good point. Unfortunately this doesn't seem easy to fix, so I'd like to leave it as follow-up.

- When you remove the width related classes, the window resize behavior becomes weird: https://youtu.be/rCnn_EhWrYs
- When you have the current classes in place, it works well: https://youtu.be/FQtzNnYi4N4

Also, FYI this is with `w-96` 😬 not quite wide enough:

<img width="600" alt="Screenshot 2021-03-26 at 09 07 38" src="https://user-images.githubusercontent.com/599268/112601936-55cdb700-8e13-11eb-88e3-81a736da89c5.png">

Reply:
Yep, that's what I meant with _futuristic_ in https://github.com/gitpod-io/gitpod/pull/3560#discussion_r601662022. Let's drop this idea for now. 😂 

Thanks for giving this a try @jankeromnes! 🏀